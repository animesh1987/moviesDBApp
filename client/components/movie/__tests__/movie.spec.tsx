import * as React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { MovieCard } from '../index';

describe('Movie Details Test', () => {
  const movie = {
    title: 'Test',
    overview: 'Test overview',
    release_date: "20-04-04",
    runtime: 144,
    poster_path: 'abc',
    genres: ['test']
  };

  const testFn = jest.fn();

  it('renders correctly', () => {
    const component = create(
      <MovieCard movie={movie} goToMovie={testFn}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});