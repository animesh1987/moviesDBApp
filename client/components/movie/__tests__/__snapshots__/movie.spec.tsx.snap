// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Test renders correctly 1`] = `
<div
  className="movie-card"
  onClick={[Function]}
>
  <div
    className="banner-image"
    style={
      Object {
        "backgroundImage": "url(abc)",
      }
    }
  />
  <div
    className="layout-row layout-align-space-between layout-align-center"
  >
    <span
      className="movie-title"
    >
      Test
    </span>
    <span
      className="release-date"
    >
      20-0
    </span>
  </div>
  <div
    className="genre-details"
  >
    test
  </div>
</div>
`;
